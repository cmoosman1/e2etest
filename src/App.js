import React, { useState } from 'react';

import UserList from './components/UserList';
import GetCoinsForm from './components/GetCoinsForm';
import { Card, Col, Container, Row } from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.min.css';

const App = () => {
  /**
   * setup hooks 
   */
  const [messages, setMessages] = useState([]);
  
  /**
   * 
   * @param {*} newMessage 
   * setMessages(newMessage) updates state from child compnent
   */
  const handleSend = newMessage => {
    setMessages(newMessage);
  };

  return (
    <Container>
      <Row>
        <Col xs={12} md={8}>
          <h4>Coin calculation:</h4>
          <GetCoinsForm onSend={handleSend} data={messages}/>
        </Col>
      </Row>
      <Row>
        <Col>
          <h4>Current users:</h4>
          <UserList />
        </Col>
      </Row>
    </Container>
  );
};

export default App;