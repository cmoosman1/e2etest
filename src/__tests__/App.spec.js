import React from 'react';
import {
   act,
  render,
  fireEvent,
  cleanup,
} from '@testing-library/react';

import App from '../App';

describe('<App />', () => {

  afterEach(cleanup);

  describe('render application', () => {
    beforeEach(() => {
      let getByTestId;
      let sendHandler = jest.fn();
      ({ getByTestId } = render(<App onSend={sendHandler} />));

      fireEvent.click(
        getByTestId('coinText'),
        {
          target: {
            value: '$7.98',
          },
        },
      );

      fireEvent.click(getByTestId('sendButton'));
    });

    it("renders", () => {
      const { asFragment } = render(<App />);
      act(() => {expect(asFragment()).toMatchSnapshot()});
    });
  });
});