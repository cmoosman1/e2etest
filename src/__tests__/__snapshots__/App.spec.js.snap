// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> render application renders 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col-md-8 col-12"
      >
        <h4>
          Coin calculation:
        </h4>
        <div>
          <div
            class="mb-3 form-font input-group"
          >
            <div
              class="input-group-prepend"
            >
              <span
                class="input-group-text"
                id="inputGroupPrepend"
              >
                $
              </span>
            </div>
            <input
              class="form-control"
              data-testid="coinText"
              placeholder="Input dollar amout to see coin conversion"
              type="text"
              value=""
            />
            <div
              class="input-group-append"
            >
              <button
                class="btn btn-outline-secondary"
                data-testid="sendButton"
                type="button"
              >
                Convert
              </button>
            </div>
          </div>
          <div />
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <h4>
          Current users:
        </h4>
        <div
          class="container"
        >
          <div
            class="row"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
